datadog_checks_base/datadog_checks/base/checks/win/wmi/__init__.py

```diff
@@ -226,48 +264,70 @@ def _submit_metrics(self, metrics, metric_name_and_type_by_property):
+        # type: (str, str, str, Any) -> str
         """
         Return an index key for a given instance. Useful for caching.
         """
         if other:
             return "{host}:{namespace}:{wmi_class}-{other}".format(
                 host=host, namespace=namespace, wmi_class=wmi_class, other=other
             )
-
         return "{host}:{namespace}:{wmi_class}".format(host=host, namespace=namespace, wmi_class=wmi_class)
 
+    def get_running_wmi_sampler(self, properties, filters, **kwargs):
+        # type: (Iterable[str], List[Dict[str, WMIFilter]], Dict) -> WMISampler
+        return self._get_running_wmi_sampler(
+            instance_key=None,
+            wmi_class=self.wmi_class,
+            properties=properties,
+            tag_by=self.tag_by,
+            filters=filters,
+            host=self.host,
+            namespace=self.namespace,
+            provider=self.provider,
+            username=self.username,
+            password=self.password,
+            **kwargs
+        )
+
     def _get_running_wmi_sampler(self, instance_key, wmi_class, properties, tag_by="", **kwargs):
+        # type: (Any, str, Iterable[str], str, Any) -> WMISampler
         """
         Return a running WMISampler for the given (class, properties).
 
         If no matching WMISampler is running yet, start one and cache it.
         """
-        properties = list(properties) + [tag_by] if tag_by else list(properties)
+        if not self._wmi_sampler:
+            properties = list(properties) + [tag_by] if tag_by else list(properties)
+            self._wmi_sampler = WMISampler(self.log, wmi_class, properties, **kwargs)
+            self._wmi_sampler.start()
 
-        if instance_key not in self.wmi_samplers:
-            wmi_sampler = WMISampler(self.log, wmi_class, properties, **kwargs)
-            wmi_sampler.start()
-            self.wmi_samplers[instance_key] = wmi_sampler
+        return self._wmi_sampler
 
-        return self.wmi_samplers[instance_key]
+    def get_wmi_properties(self):
+        # type: () -> WMIProperties
+        return self._get_wmi_properties(None, self.metrics_to_capture, self.tag_queries)
 
     def _get_wmi_properties(self, instance_key, metrics, tag_queries):
+        # type: (Any, List[List[str]], List[List[str]]) -> WMIProperties
         """
         Create and cache a (metric name, metric type) by WMI property map and a property list.
         """
-        if instance_key not in self.wmi_props:
+        if not self._wmi_props:
             metric_name_by_property = dict(
                 (wmi_property.lower(), (metric_name, metric_type)) for wmi_property, metric_name, metric_type in metrics
-            )
-            properties = map(lambda x: x[0], metrics + tag_queries)
-            self.wmi_props[instance_key] = (metric_name_by_property, properties)
+            )  # type: Dict[str, Tuple[str, str]]
+            properties = map(lambda x: x[0], metrics + tag_queries)  # type: Iterable[str]
```

Comment:
Any thoughts on making this an actual list? None of `metrics` or `tag_queries` are lazy, so it looks like we don't need the iterator API provided by `map`.

```suggestion
            properties = [item[0] for item in metrics + tag_queries]
```

I get that this is an unrelated change so we could issue it as a separate PR if necessary.

Reply:
I don't want to change behaviour